gravitee-apim-console-webui/src/management/settings/api-portal-header/migrated/api-portal-header.component.ts

```diff
@@ -0,0 +1,243 @@
+    this.unsubscribe$.unsubscribe();
+  }
+
+  private initSettings(): void {
+    this.portalSettingsService
+      .get()
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe((settings: PortalSettings) => {
+        this.settings = settings;
+      });
+  }
+
+  private initListSubscription() {
+    this.environmentApiHeadersService
+      .getHeadersList$()
+      .pipe(
+        catchError(_ => {
+          this.snackBarService.error('Error occurred.');
+          return [];
+        }),
+        takeUntil(this.unsubscribe$)
+      )
+      .subscribe((list: ApiPortalHeader[]): void => {
+        this.headersList = list;
+      });
+  }
+
+  private createMessage(option: ToggleOptions): string {
+    const messages = {
+      [ToggleOptions.show_tags]: 'Tags are now ' + (this.settings.portal.apis.apiHeaderShowTags.enabled ? 'visible' : 'hidden'),
+      [ToggleOptions.show_categories]: 'Categories are now ' + (this.settings.portal.apis.apiHeaderShowCategories.enabled ? 'visible' : 'hidden'),
+      [ToggleOptions.promoted_api_mode]: 'Promoted API is now ' + (this.settings.portal.apis.promotedApiMode.enabled ? 'visible' : 'hidden')
+    };
+    return messages[option];
+  }
+
+
+  public saveToggle(toggleOption: ToggleOptions): void {
+    this.portalSettingsService.get()
+      .pipe(
+        switchMap((currentSettings: PortalSettings) => {
+          return this.portalSettingsService.save(
+            { ...currentSettings, portal: { ...currentSettings.portal, apis: this.settings.portal.apis } });
+        }),
+        tap(() => {
+          this.snackBarService.success(this.createMessage(toggleOption));
+        }),
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$)
+      )
+      .subscribe((updatedSettings: PortalSettings) => {
+        this.settings = updatedSettings;
+      });
+  };
+
+  public isReadonlySetting(property: string): boolean {
+    return PortalSettingsService.isReadonly(this.settings, property);
+  };
+
+  private getList(): void {
+    this.environmentApiHeadersService
+      .getApiHeaders()
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe();
+  }
+
+  public moveUp(apiPortalHeader: ApiPortalHeader): void {
+    const updatedHeader = { ...apiPortalHeader, order: apiPortalHeader.order - 1 };
+    this.environmentApiHeadersService
+      .updateApiHeader(updatedHeader)
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe();
+  }
+
+  public moveDown(apiPortalHeader: ApiPortalHeader): void {
+    const updatedHeader = { ...apiPortalHeader, order: apiPortalHeader.order + 1 };
+    this.environmentApiHeadersService
+      .updateApiHeader(updatedHeader)
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe();
+  }
+
+  public createHeader(): void {
+    this.matDialog
+      .open<HeaderDialogComponent, HeaderDialogData, HeaderDialogData>(HeaderDialogComponent, {
```

Comment:
And use HeaderDialogResult here 
```suggestion
      .open<HeaderDialogComponent, HeaderDialogData, HeaderDialogResult>(HeaderDialogComponent, {
```

It's a convention that we try to respect on our dialogues. to type correctly

Reply:
.open<HeaderDialogComponent, HeaderDialogData, HeaderDialogResult>,
the component gets the same type and returns in result. So , if I change HeaderDialogData => HeaderDialogResult,
the function types would look: 
.open<HeaderDialogComponent, HeaderDialogResult, HeaderDialogResult>

Reply:
if it's the same type, it's better to indicate in the component. A data is not a result.
`export type HeaderDialogData = HeaderDialogResult`

So all dialogue always has these 2 type ...Data ...Result

Reply:
done() :-)